### PR TITLE
Nrp fix2

### PR DESCRIPTION
**Pull request by kybeak**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md).**
- [ ] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md).**
- [ ] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.
- [ ] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [ ] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change).
- [ ] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [ ] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [ ] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [ ] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

File: setup/azurecmdfiles.wxi [hunk 1]
```diff
@@ -1858,9 +1858,6 @@
                         <Component Id="cmp8D637393B06DDB4E95F9C9DE02FBDC88" Guid="*">
                             <File Id="fil2E0A33EC74AA6C08679F32E280E41EC3" KeyPath="yes" Source="$(var.sourceDir)\ResourceManager\AzureResourceManager\AzureRM.Profile\Microsoft.Threading.Tasks.Extensions.dll" />
                         </Component>
-                        <Component Id="cmpD70748C6F934F291B41467868246436F" Guid="*">
-                            <File Id="fil61F9393E7F2BF00B994B5674A27808B6" KeyPath="yes" Source="$(var.sourceDir)\ResourceManager\AzureResourceManager\AzureRM.Profile\Microsoft.Web.Deployment.dll" />
-                        </Component>
```
Comment (bganapa) [inline setup/azurecmdfiles.wxi]: We should not be removing this

File: setup/azurecmdfiles.wxi [hunk 2]
```diff
@@ -4470,15 +4452,6 @@
                         <Component Id="cmp71530FA712EAD3B1A8C2567DB3B9DC17" Guid="*">
                             <File Id="fil125F24B9366F55DB3D98DA60DD2D5F42" KeyPath="yes" Source="$(var.sourceDir)\ServiceManagement\Azure\Sql\Microsoft.Threading.Tasks.Extensions.Desktop.dll" />
                         </Component>
-                        <Component Id="cmp3343A6EBDBAC93AE47AF05414FB13AA1" Guid="*">
```
Comment (bganapa) [inline setup/azurecmdfiles.wxi]: there is no need to change the wxi file for your changes

File: src/ResourceManager/Compute/Commands.Compute.Test/packages.config [hunk 3]
```diff
@@ -7,7 +7,7 @@
   <package id="Microsoft.Azure.Graph.RBAC" version="1.9.0-preview" targetFramework="net45" />
   <package id="Microsoft.Azure.Management.Authorization" version="2.0.0" targetFramework="net45" />
   <package id="Microsoft.Azure.Management.Compute" version="11.3.1-prerelease" targetFramework="net45" />
-  <package id="Microsoft.Azure.Management.Network" version="3.1.0-preview" targetFramework="net45" />
```
Comment (bganapa) [inline src/ResourceManager/Compute/Commands.Compute.Test/packages.config]: We will need to update this with a new package version, after we publish the SDK changes

File: src/ResourceManager/Network/Commands.Network.Test/Commands.Network.Test.csproj [hunk 4]
```diff
@@ -310,6 +311,15 @@
     <None Include="SessionRecords\Commands.Network.Test.ScenarioTests.TestDnsAvailabilityTest\TestDnsAvailability.json">
       <CopyToOutputDirectory>Always</CopyToOutputDirectory>
     </None>
+    <None Include="SessionRecords\Commands.Network.Test.ScenarioTests.VirtualNetworkGatewayConnectionTests\TestVirtualNetworkeExpressRouteGatewayConnectionCRUD.json">
```
Comment (bganapa) [inline src/ResourceManager/Network/Commands.Network.Test/Commands.Network.Test.csproj]: Why are these files getting added? I see only one new test

File: setup/azurecmdfiles.wxi (hunk 1 above)
Comment (kybeak) [inline setup/azurecmdfiles.wxi]: because we can't even compile with this changes.

File: setup/azurecmdfiles.wxi (hunk 2 above)
Comment (kybeak) [inline setup/azurecmdfiles.wxi]: same at here.

File: src/ResourceManager/Compute/Commands.Compute.Test/packages.config (hunk 3 above)
Comment (kybeak) [inline src/ResourceManager/Compute/Commands.Compute.Test/packages.config]: yes I will update it

File: src/ResourceManager/Network/Commands.Network.Test/Commands.Network.Test.csproj (hunk 4 above)
Comment (kybeak) [inline src/ResourceManager/Network/Commands.Network.Test/Commands.Network.Test.csproj]: because we are cherry picking the fix of others.

**Comment by markcowl** [comment]
@azuresdkci add to whitelist

**Comment by cormacpayne** [comment]
@kybeak there seems to be an error with the version of `ClientRuntime` you are using that is causing the build to fail

**Comment by bganapa** [comment]
The error should go away once we consume 3.1.1 version of the sdk

Sent from my Windows Phone
________________________________

**Comment by kybeak** [comment]
yes, we were planning to update the SDK version.
Updated PR to cherry pick from Bala's change.

File: src/ResourceManager/Compute/Commands.Compute.Test/packages.config (hunk 3 above)
Comment (bganapa) [inline src/ResourceManager/Compute/Commands.Compute.Test/packages.config]: This got updated

**Comment by bganapa** [comment]
@azuresdkci test this

**Comment by markcowl** [comment]
@azuresdkci retest this please

**Comment by bganapa** [comment]
@azuresdkci retest this

**Pull request merged**